prov/sockets/src/sock_progress.c

```diff
@@ -1774,7 +1774,7 @@ static int sock_pe_read_hdr(struct sock_pe *pe, struct sock_rx_ctx *rx_ctx,
 
 	msg_hdr = &pe_entry->msg_hdr;
 	if (sock_pe_peek_hdr(pe, pe_entry))
-		return 0;
+		return -1;
```

Comment:
Sure, just a little bit more code. Ideally, there should be a named constant helping understand what's going on, but now (-1) is used consistently throughout the code.